G1ANT.Addon.Azure/AzureManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.Azure.KeyVault;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
+using System.Threading.Tasks;
+
+namespace G1ANT.Addon.Azure
+{
+    public class AzureHelper
+    {
+        private static string _clientId;
+        private static string _clientSecret;
+
+        private static async Task<string> GetToken(string authority, string resource, string scope)
+        {
+            var authContext = new AuthenticationContext(authority);
+            var clientCred = new ClientCredential(_clientId, _clientSecret);
+            var result = await authContext.AcquireTokenAsync(resource, clientCred);
+
+            if (result == null)
+            {
+                throw new InvalidOperationException("Failed to obtain the JWT token");
+            }
+
+            return result.AccessToken;
+        }
+
+        public static string GetSecret(string clientId, string clientSecret, Uri keyVaultUri, string secretName)
+        {
```

Comment:
They are validated in the implementation of the AzureKeyVault

Reply:
Ain't that a bit too late?

Reply:
Don't resolve conversation without clarification.
IMHO clientId and clientSecret should be initialized at manager constructor (since they define basic object state) and that is the place where the values should be validated.